### PR TITLE
[EPT-1709] Add a component for hover options and function for editor URL

### PR DESCRIPTION
**Pull request by shanthru**
[EPT-1709](https://sailthru.atlassian.net/browse/EPT-1709)

This PR:
Adds a component for the hover options on the page table.
Adds a function for the redirect to the editor URL for HTML or Visual pages.

relies on PHP branch: https://github.com/sailthru/php/pull/5229

Note: npm audit fix was run to resolve issues

### CODE REVIEW

File: src/ListView/PageTable.jsx [hunk 1]
```diff
@@ -132,3 +110,13 @@ function PageTable({ loading, pages, getPageById, displayModal }) {
 }
 
 export { PageTable };
+
+export function pageEditorUrl(pageId, pageMode) {
+  const htmlPageUrl = `/page?page_id=${pageId}`; // TODO add click through see EPT-1713
+  const visualPageUrl = "/hosted-pages-list";
```
Comment (jdrury-sailthru) [inline src/ListView/PageTable.jsx:116]: nit: these urls should come from the uiapi, IMO

File: src/ListView/PageTableHoverOptions.jsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  ];
+
+  return (
+    <HoverDisclosure disclosureWidth="150px">
+      <OptionsMenu options={clickOptions} />
+    </HoverDisclosure>
+  );
+};
+
+export default PageTableHoverOptions;
```
Comment (jdrury-sailthru) [inline src/ListView/PageTableHoverOptions.jsx:32]: nit: be consistent with named exports vs default exports

File: src/ListView/PageTableHoverOptions.jsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import { HoverDisclosure, OptionsMenu } from "@sailthru/stui-components";
+
+const PageTableHoverOptions = ({ pageId, pageEditorUrl }) => {
+  const clickOptions = [
+    {
+      label: "Edit",
+      icon: "fal fa-edit fa-fw",
+      iconHover: "fas fa-edit fa-fw",
+      onClick: () => window.open(`${pageEditorUrl}`, "_self"), // TODO see EPT-1713 (add routing?)
```
Comment (jdrury-sailthru) [inline src/ListView/PageTableHoverOptions.jsx:10]: question: what is the self option for?

**Review by jdrury-sailthru — APPROVED** [review]
+1 for tidiness

File: src/ListView/PageTable.jsx (hunk 1 above)
Comment (shanthru) [reply src/ListView/PageTable.jsx]: I've moved this to uiapi for now
https://github.com/sailthru/php/pull/5229

Per discussion, we should consider using the same editor url when adding a visual editor but render the app according to whether the page mode is visual or html.

File: src/ListView/PageTableHoverOptions.jsx (hunk 2 above)
Comment (shanthru) [reply src/ListView/PageTableHoverOptions.jsx]: Updated!

File: src/ListView/PageTableHoverOptions.jsx (hunk 3 above)
Comment (shanthru) [reply src/ListView/PageTableHoverOptions.jsx:10]: opens the html editor in the same tab instead of a new tab

**Review by lindsaycooper — APPROVED** [review]
Approving based on James's approval above

File: src/ListView/PageTableHoverOptions.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    {
+      label: "Delete",
+      icon: "fal fa-trash fa-fw",
+      iconHover: "fal fa-trash fa-fw",
```
Comment (lindsaycooper) [inline src/ListView/PageTableHoverOptions.jsx:15]: Does this need to be updated to make the hover a bolder icon?
```suggestion
      iconHover: "fas fa-trash fa-fw",
```

**Review by lindsaycooper — APPROVED** [review]
QA Steps: One minor visual comment needs addressing (Delete icon doesn't bold)

- [x] http://my.sailthru-sb.com/hosted-pages-list loads the table
    image
- [x] Clicking the name of the page redirects to the HTML editor with the correct page id
- [x] Clicking the preview icon in the URL opens a new tab with the page preview
- [x] Clicking "Edit" redirects to the HTML editor with the correct page id
- [x] Clicking "Delete" prints a console log with the correct page id
- [x] Last Modified date updates when HTML editor saves an update
- [x] New HTML pages show in the list when they are created in the HTML editor

- [x] Hover over the 3 dots shows edit and delete options
    image

- [x] Hover over Edit option highlights option + bold icon
    image

- [ ] Hover over Delete option highlights option, **but doesn't change to a bold icon**
    image

**Comment by shanthru** [comment]
> * [ ]  Hover over Delete option highlights option, **but doesn't change to a bold icon**
>   image

This has been updated. Thanks!

File: src/ListView/PageTableHoverOptions.jsx (hunk 4 above)
Comment (shanthru) [reply src/ListView/PageTableHoverOptions.jsx]: Yes! updated

File: src/ListView/PageTable.jsx [hunk 5]
```diff
@@ -132,3 +110,13 @@ function PageTable({ loading, pages, getPageById, displayModal }) {
...
+  if (pageMode == "visual") {
+    return visualPageUrl;
+  } else {
+    return htmlPageUrl;
+  }
```
[the remark this answers is not in the document]
Comment (shanthru) [reply src/ListView/PageTable.jsx]: this is refactored to uiapi

**Comment by lindsaycooper** [comment]
> 

Looks good! QA +1

**Pull request merged**